### ISSUE TITLE
Export inference model from ckpt files

### ISSUE THREAD

**Issue by abdou31** [yinguobing/cnn-facial-landmark#53]
Hello Sir,
I have trained my own dataset, I want to generate frozen_inference_graph.pb from ckpt files,
How can I do that?
I don't know the names of the node,
Can you tell me what are the nodes that you are used to freezing the model and producing .pb file?
I need help.
Updating: I use a script to do this and I set output nodes name "logits/BiasAdd".
I get frozen_inference_graph with size 28 Mo.
I want to know is it correct and logical to get a .pb frozen graph with this size?
Thanks

**Comment by yinguobing** [comment]
Sorry for the late response.

There are many ways to export model for inference. For apps in the cloud the `SavedModel` format is recommended as TensorFlow Serving could use this format directly. For other apps running on PC/Mobile/Embedded devices, GraphDef is preferred since it only requires a single file for the whole model.

The latest repo uses SavedModel as the default output format. If you want to export the model in a single GraphDef file, you are probably doing "model freezing".

TensorFlow has officially provided a python module to do this freezing thing:
https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/tools/freeze_graph.py

You only need to provide the model either in checkpoint files or SavedModel format, output node name, and the desired name for frozen model, then, it's done.

If you are not sure what the output node name is, you can check the model architecture in TensorBoard when training files (event files) are available. If you only have the SavedModel, it's also possible through [importing the SavedModel to TensorBoard](https://yinguobing.com/view-tensorflow-graph-from-saved-model/).

Wish this helps.

**Comment by abdou31** [comment]
Thanks.
I think the model is correct because I test `landmark_video.py` with the frozen inference and I have got an acceptable result.
